### ISSUE TITLE
sucessfully configure, but build error with bazel

### ISSUE THREAD

**Issue by Kunge** [tensorflow/tensorflow#5191]
the configure sucessfully finished
but when run "bazel build -c opt //tensorflow/tools/pip_package:build_pip_package"
it reports the error: ......./tensorflow/tensorflow/python/BUILD:1806:1: in cc_library rule //tensorflow/python:tf_session_helper: non-test target '//tensorflow/python:tf_session_helper' depends on testonly target '//tensorflow/python:construction_fails_op' and doesn't have testonly attribute set

**Comment by Kunge** [comment]
it works when comment all the testonly dependencies

**Comment by drpngx** [comment]
#5093 

Could you tell us more about your configuration?

**Comment by drpngx** [comment]
@davidzchen @jart 

The comment before "construction_fails_op" makes it clear that it's a genuine `testonly`.

I can build just fine, but:

```
bazel query 'allpaths(//tensorflow/python:tf_session_helper, //tensorflow/python:construction_fails_op)'
//tensorflow/python:tf_session_helper
//tensorflow/python:construction_fails_op
```

And also some other test ops?

```
bazel query 'deps(//tensorflow/python:tf_session_helper) intersect //tensorflow/python/...'
//tensorflow/python:tf_session_helper
//tensorflow/python:test_ops_kernels
//tensorflow/python:numpy_lib
//tensorflow/python:construction_fails_op
```

I can't quite see why it should be the case.

**Comment by Kunge** [comment]
I comment these two lines of "..../python/BUILD":
1811 ":construction_fails_op", 
1813         ":test_ops_kernels",
then the build is fine.

**Comment by drpngx** [comment]
Did you test the rest of the code? It probably fails somewhere. If you can fix the code, then it would be good to send a PR

**Comment by jart** [comment]
We've got cl/137268906 in the oven which fixes this internally. It's also being cherry-picked onto r0.11 in #5209.

**Comment by drpngx** [comment]
Nice thanks!

**Comment by davidzchen** [comment]
Thanks!

**Comment by Ingwar** [comment]
I've got the same error when I attempted to build the last release (https://github.com/tensorflow/tensorflow/releases/tag/v0.11.0). Probably, the fix was not included into it.

**Comment by jart** [comment]
Re-opening this issue for triage.

**Comment by drpngx** [comment]
How about 0.11rc2?

**Comment by Ingwar** [comment]
The master branch builds normally.

**Comment by girving** [comment]
@jart Is this fixed now that that CL is in?

**Comment by jkschin** [comment]
Just tried v0.11.0. Doesn't work. Seems to work on master and it's building now.

**Comment by gunan** [comment]
This issue should be resolved in both master and 0.12
